<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>624</width>
    <height>481</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Current class:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="group"/>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>&amp;Manage students and classes</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="pushButton">
            <property name="text">
             <string>Create new class</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_2">
            <property name="text">
             <string>Remove this class</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Edit students in the current class:</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QListWidget" name="students"/>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QPushButton" name="pushButton_3">
              <property name="text">
               <string>&amp;Add
new
student</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_4">
              <property name="text">
               <string>&amp;Remove
selected
student</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Double-click on a student to edit the name</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>&amp;Generate subgroups or pairs</string>
       </attribute>
       <widget class="QListWidget" name="attendants">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>491</width>
          <height>91</height>
         </rect>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::ExtendedSelection</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label_4">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>10</y>
          <width>181</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Mark the present students:</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_5">
        <property name="geometry">
         <rect>
          <x>240</x>
          <y>10</y>
          <width>101</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>&amp;All present</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_6">
        <property name="geometry">
         <rect>
          <x>372</x>
          <y>10</y>
          <width>111</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>&amp;None present</string>
        </property>
       </widget>
       <widget class="QLabel" name="absenseStats">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>140</y>
          <width>311</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>%d students present, %d students absent</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_7">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>170</y>
          <width>171</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>&amp;Split to subgroups of N:</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_8">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>200</y>
          <width>171</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>S&amp;plit to N subgroups:</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="desiredSGsize">
        <property name="geometry">
         <rect>
          <x>190</x>
          <y>170</y>
          <width>91</width>
          <height>24</height>
         </rect>
        </property>
        <property name="text">
         <string>2</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="desiredSGcount">
        <property name="geometry">
         <rect>
          <x>190</x>
          <y>200</y>
          <width>91</width>
          <height>24</height>
         </rect>
        </property>
        <property name="text">
         <string>3</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioPreferExtended">
        <property name="geometry">
         <rect>
          <x>300</x>
          <y>170</y>
          <width>211</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Prefer incomplete subgroups</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioButton_2">
        <property name="geometry">
         <rect>
          <x>300</x>
          <y>200</y>
          <width>201</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Prefer extended subgroups</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="Line" name="line">
        <property name="geometry">
         <rect>
          <x>280</x>
          <y>140</y>
          <width>20</width>
          <height>81</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label_5">
        <property name="geometry">
         <rect>
          <x>330</x>
          <y>140</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>In case of odd split:</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_6">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>240</y>
          <width>57</width>
          <height>14</height>
         </rect>
        </property>
        <property name="text">
         <string>Result:</string>
        </property>
       </widget>
       <widget class="QTableWidget" name="result">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>260</y>
          <width>571</width>
          <height>111</height>
         </rect>
        </property>
        <property name="dragEnabled">
         <bool>true</bool>
        </property>
        <property name="dragDropOverwriteMode">
         <bool>false</bool>
        </property>
        <property name="dragDropMode">
         <enum>QAbstractItemView::DragDrop</enum>
        </property>
        <property name="defaultDropAction">
         <enum>Qt::IgnoreAction</enum>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>624</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExport_to_file"/>
    <addaction name="actionImport_from_file"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionExport_to_file">
   <property name="text">
    <string>Export to file</string>
   </property>
  </action>
  <action name="actionImport_from_file">
   <property name="text">
    <string>Import from file</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
